schemas/hook-status.yml

```diff
@@ -0,0 +1,55 @@
+$schema: http://json-schema.org/draft-04/schema#
+title:   "Hook status response"
+description: |
+  A snapshot of the current status of a hook.
+type: object
+properties:
+  lastFire:
+    description: |
+      Information about the last time this hook fired.  This property is only present
+      if the hook has fired at least once.
+    type: object
+    oneOf:
+      - title: Successful Fire
+        description: "Information about a successful firing of the hook"
+        type: object
+        properties:
+          taskId:
```

Comment:
Other places we've used `oneOf` we've often added a property:
`kind: enum: ['task-fired']` and `kind: enum: ['fire-failure']`

to each of them... So it's that when reading the object you don't have to test for property presence, rather you look at the `kind` property. Our golang binding doesn't take advantage of this yet, but I hope someday we will (or we'll find another way of doing it).

It's somewhat nice to have this even in Javascript.